src/app/pages-block/pages-block.component.ts

```diff
@@ -1,45 +1,100 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Output, EventEmitter, OnDestroy } from '@angular/core';
 import { ICoursePage } from '../interfaces/course.interface';
 import { CoursesService } from './services/courses.service';
+import { debounceTime, distinctUntilChanged, filter, fromEvent, map, Subject, takeUntil } from "rxjs";
+import { LoadingService } from '../shared/loading-block/servises/loading.service';
+
 
 @Component({
   selector: 'app-pages-block',
   templateUrl: './pages-block.component.html',
   styleUrls: ['./pages-block.component.scss'],
 })
 
-export class PagesBlockComponent implements OnInit {
+export class PagesBlockComponent implements OnInit, OnDestroy {
 
   @Output() addButtonClicked: EventEmitter<void> = new EventEmitter()
 
+  showLoadMore: boolean = true;
   courses: ICoursePage[] = [];
+  numberOfCourses: number = 3;
+  private currentSubscribes: Subject<void> = new Subject<void>();
 
-  constructor(private coursesPagesService: CoursesService) { }
+  constructor(private coursesPagesService: CoursesService, private loadingService: LoadingService) { }
 
   ngOnInit(): void {
-    this.courses = this.coursesPagesService.getCoursesList()
+
+    setTimeout(() => this.refreshCourse(), 0);
+    this.searchFunction();
+  }
+
+  searchFunction(): void {
+    const searchBox = document.getElementById('search-box') as HTMLInputElement;
+    if (searchBox) {
+      let searchData$ = fromEvent(searchBox, 'input').pipe(
```

Comment:
I believe this variable is redundant
start with "fromevent" and don't separate takeUntil from other operators